File: src/Domain/A.Domain/Models/Customer.cs
using System.ComponentModel.DataAnnotations;

namespace A.Domain.Models
{
    public class Buyer : BaseEntity
    {
        public string? Name { get; set; }
        
        public string? LoyaltyCard { get; set; }

        public Address Address { get; set; }

        public Buyer()
        {
            Address = new Address();
        }
    }
}

File: src/Presentation/A.Presentation/ClientApp/src/app/models/customer.ts
import { IBaseEntity } from "./baseEntity";

export interface IAddress {
  Id?: string;
  Unit?: string;
  Street?: string;
  City?: string;
  State?: string;
  Country?: string;
  ZipCode?: string;
}


export interface IBuyer extends IBaseEntity {
  Name?: string;
  LoyaltyCard?: string;
  Address?: IAddress;
}
